edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/AuthAgentStartup.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using Microsoft.AspNetCore.Builder;
+
+    public class AuthAgentStartup
+    {
+        readonly AuthAgentProtocolHeadConfig config;
+
+        public AuthAgentStartup(AuthAgentProtocolHeadConfig config)
+        {
+            this.config = config;
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    "authenticate",
+                    this.config.BaseUrl,
+                    defaults: new { controller = "AuthAgent", action = "HandleAsync" });
```

Comment:
Should we use routing attribute instead?

Reply:
I wanted to be sure that the authentication doesn't handle any other requests (=to pick up classes with appropriate attributes automatically), also it uses configuration dependent parts (the baseurl) which cannot be done with routing attributes.